src/nvim/CMakeLists.txt

```diff
@@ -11,31 +11,38 @@ endif()
 endif()
 
 set(TOUCHES_DIR ${PROJECT_BINARY_DIR}/touches)
+set(GENERATOR_DIR ${CMAKE_CURRENT_LIST_DIR}/generators)
 set(GENERATED_DIR ${PROJECT_BINARY_DIR}/src/nvim/auto)
-set(MSGPACK_GENERATOR ${PROJECT_SOURCE_DIR}/scripts/genmsgpack.lua)
+set(API_DISPATCH_GENERATOR ${GENERATOR_DIR}/gen_api_dispatch.lua)
+set(UI_EVENTS_GENERATOR ${GENERATOR_DIR}/gen_ui_events.lua)
```

Comment:
since I requested another change in this file I'll suggest this bikeshed:

    set(API_UI_EVENTS_GENERATOR ${GENERATOR_DIR}/gen_api_ui_events.lua)